source/common/grpc/status.cc

```diff
@@ -85,5 +87,48 @@ uint64_t Utility::grpcToHttpStatus(Status::GrpcStatus grpc_status) {
   }
 }
 
+std::string Utility::grpcStatusToString(Status::GrpcStatus grpc_status) {
+  switch (grpc_status) {
+  case Status::WellKnownGrpcStatus::Ok:
+    return "OK";
+  case Status::WellKnownGrpcStatus::Canceled:
+    return "Canceled";
+  case Status::WellKnownGrpcStatus::Unknown:
+    return "Unknown";
+  case Status::WellKnownGrpcStatus::InvalidArgument:
+    return "InvalidArgument";
+  case Status::WellKnownGrpcStatus::DeadlineExceeded:
+    return "DeadlineExceeded";
+  case Status::WellKnownGrpcStatus::NotFound:
+    return "NotFound";
+  case Status::WellKnownGrpcStatus::AlreadyExists:
+    return "AlreadyExists";
+  case Status::WellKnownGrpcStatus::PermissionDenied:
+    return "PermissionDenied";
+  case Status::WellKnownGrpcStatus::ResourceExhausted:
+    return "ResourceExhausted";
+  case Status::WellKnownGrpcStatus::FailedPrecondition:
+    return "FailedPrecondition";
+  case Status::WellKnownGrpcStatus::Aborted:
+    return "Aborted";
+  case Status::WellKnownGrpcStatus::OutOfRange:
+    return "OutOfRange";
+  case Status::WellKnownGrpcStatus::Unimplemented:
+    return "Unimplemented";
+  case Status::WellKnownGrpcStatus::Internal:
+    return "Internal";
+  case Status::WellKnownGrpcStatus::Unavailable:
+    return "Unavailable";
+  case Status::WellKnownGrpcStatus::DataLoss:
+    return "DataLoss";
+  case Status::WellKnownGrpcStatus::Unauthenticated:
+    return "Unauthenticated";
+  case Status::WellKnownGrpcStatus::InvalidCode:
+    return "InvalidCode";
+  default:
+    return EMPTY_STRING;
```

Comment:
Just convert the number to string in the case of custom gRPC status is used?

Reply:
@lizan Yes, It is required. But if we allow this functionality, we need to change api, aren't we? For example, we must define when grpc-status is 23 then envoy must output "custom error" in bootstrap.

Reply:
I meant just print it as number when envoy doesn't know. i.e. "23"

Reply:
Maybe, users should be able to change whether they allow the custom status as an option. So we need to add `GRPC_STATUS_ALLOW_USER_DEFINED`.

Reply:
I don't see the reason we need two options here, we don't need `InvalidCode` but just simply fallback to number.